include/boost/iterator/detail/facade_iterator_category.hpp

```diff
@@ -139,17 +138,17 @@ struct iterator_category_with_traversal
     // Make sure this isn't used to build any categories where
     // convertibility to Traversal is redundant.  Should just use the
     // Category element in that case.
-    BOOST_MPL_ASSERT_NOT((
-        is_convertible<
+    BOOST_STATIC_ASSERT(
```

Comment:
Single parentheses requires variadic macros, and we are not requiring C++11. At least, not yet and not as part of this PR. Please keep the double parentheses. Here, and everywhere else where there is a comma in the condition.

Reply:
It looks like they were only required for `BOOST_MPL_ASSERT_NOT` [as documented](https://www.boost.org/doc/libs/1_82_0/libs/mpl/doc/refmanual/assert.html) and CI passes even for GCC 4.4 using C++98. I also verified that on [compiler explorer](https://godbolt.org/z/5Erz9KEh5) So if you don't feel strongly about those parentheses I'd keep the single ones especially as the extra parens would appear in the static assert output (see the godbolt link above)

Reply:
gcc supports variadic macros in C++03 as a non-standard extension, but I would like not to rely on it. Please, keep the double parenthesis.

Reply:
Ok, done